src/main/java/bio/terra/service/dataset/flight/ingest/IngestBulkMapResponseStep.java

```diff
@@ -0,0 +1,57 @@
+package bio.terra.service.dataset.flight.ingest;
+
+import bio.terra.model.BulkLoadArrayResultModel;
+import bio.terra.model.BulkLoadFileResultModel;
+import bio.terra.model.BulkLoadResultModel;
+import bio.terra.service.load.LoadService;
+import bio.terra.service.load.flight.LoadMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.StepResult;
+import java.util.List;
+import java.util.UUID;
+import java.util.function.Predicate;
+
+// It expects the following working map data:
+// - LOAD_ID - load id we are working on
+//
+public class IngestBulkMapResponseStep extends SkippableStep {
+
+  private final LoadService loadService;
+  private final String loadTag;
+
+  public IngestBulkMapResponseStep(
+      LoadService loadService, String loadTag, Predicate<FlightContext> skipCondition) {
+    super(skipCondition);
+    this.loadService = loadService;
+    this.loadTag = loadTag;
+  }
+
+  @Override
+  public StepResult doSkippableStep(FlightContext context) {
+    FlightMap workingMap = context.getWorkingMap();
+    String loadIdString = workingMap.get(LoadMapKeys.LOAD_ID, String.class);
+    UUID loadId = UUID.fromString(loadIdString);
+
+    BulkLoadArrayResultModel result = makeLoadResult(loadId, context);
+    workingMap.put(IngestMapKeys.BULK_LOAD_RESULT, result);
+
+    return StepResult.getStepResultSuccess();
+  }
+
+  @Override
+  public StepResult undoSkippableStep(FlightContext context) {
+    return StepResult.getStepResultSuccess();
```

Comment:
It looks like most cases of `SkippableStep` have no undo operation. Instead of making `undoSkippableStep()` `abstract` you could provide an implementation that returns success to avoid having to override it in every subclass.

Reply:
Great improvement!